### ISSUE TITLE
Error 500 after signin to admin console: NullPointerException 

### ISSUE THREAD

**Issue by janinko** [keycloak/keycloak#20332]
### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the latest release

### Area

admin/ui

### Describe the bug

I tried to login to the /admin console via web and got this error:

```
Unexpected Application Error!
Network response was not OK.
Error: Network response was not OK.
    at fetchWithError (https://my.server.com/resources/i3n4n/admin/keycloak.v2/assets/index-d2c499e7.js:67:43710)
    at async Agent.requestWithParams (https://my.server.com/resources/i3n4n/admin/keycloak.v2/assets/index-d2c499e7.js:67:46301)
```

```
2023-05-12 19:16:02,846 ERROR [org.keycloak.services.error.KeycloakErrorHandler] (executor-thread-5) Uncaught server error: java.lang.NullPointerException: Cannot invoke "org.keycloak.models.RoleModel.getId()" because "role" is null
	at org.keycloak.models.cache.infinispan.UserAdapter.hasRole(UserAdapter.java:357)
	at org.keycloak.services.resources.admin.AdminConsole.whoAmI(AdminConsole.java:230)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:170)
	at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:130)
	at org.jboss.resteasy.core.ResourceMethodInvoker.internalInvokeOnTarget(ResourceMethodInvoker.java:660)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTargetAfterFilter(ResourceMethodInvoker.java:524)
	at org.jboss.resteasy.core.ResourceMethodInvoker.lambda$invokeOnTarget$2(ResourceMethodInvoker.java:474)
	at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:364)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTarget(ResourceMethodInvoker.java:476)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:434)
	at org.jboss.resteasy.core.ResourceLocatorInvoker.invokeOnTargetObject(ResourceLocatorInvoker.java:192)
	at org.jboss.resteasy.core.ResourceLocatorInvoker.invoke(ResourceLocatorInvoker.java:141)
	at org.jboss.resteasy.core.ResourceLocatorInvoker.invoke(ResourceLocatorInvoker.java:32)
	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:492)
	at org.jboss.resteasy.core.SynchronousDispatcher.lambda$invoke$4(SynchronousDispatcher.java:261)
	at org.jboss.resteasy.core.SynchronousDispatcher.lambda$preprocess$0(SynchronousDispatcher.java:161)
	at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:364)
	at org.jboss.resteasy.core.SynchronousDispatcher.preprocess(SynchronousDispatcher.java:164)
	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:247)
	at io.quarkus.resteasy.runtime.standalone.RequestDispatcher.service(RequestDispatcher.java:73)
	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.dispatch(VertxRequestHandler.java:151)
	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.handle(VertxRequestHandler.java:82)
	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.handle(VertxRequestHandler.java:42)
	at io.vertx.ext.web.impl.RouteState.handleContext(RouteState.java:1284)
	at io.vertx.ext.web.impl.RoutingContextImplBase.iterateNext(RoutingContextImplBase.java:173)
	at io.vertx.ext.web.impl.RoutingContextImpl.next(RoutingContextImpl.java:140)
	at io.quarkus.vertx.http.runtime.StaticResourcesRecorder$2.handle(StaticResourcesRecorder.java:84)
	at io.quarkus.vertx.http.runtime.StaticResourcesRecorder$2.handle(StaticResourcesRecorder.java:71)
	at io.vertx.ext.web.impl.RouteState.handleContext(RouteState.java:1284)
	at io.vertx.ext.web.impl.RoutingContextImplBase.iterateNext(RoutingContextImplBase.java:173)
	at io.vertx.ext.web.impl.RoutingContextImpl.next(RoutingContextImpl.java:140)
	at io.quarkus.vertx.http.runtime.VertxHttpRecorder$6.handle(VertxHttpRecorder.java:430)
	at io.quarkus.vertx.http.runtime.VertxHttpRecorder$6.handle(VertxHttpRecorder.java:408)
	at io.vertx.ext.web.impl.RouteState.handleContext(RouteState.java:1284)
	at io.vertx.ext.web.impl.RoutingContextImplBase.iterateNext(RoutingContextImplBase.java:173)
	at io.vertx.ext.web.impl.RoutingContextImpl.next(RoutingContextImpl.java:140)
	at org.keycloak.quarkus.runtime.integration.web.QuarkusRequestFilter.lambda$createBlockingHandler$0(QuarkusRequestFilter.java:82)
	at io.quarkus.vertx.core.runtime.VertxCoreRecorder$14.runWith(VertxCoreRecorder.java:576)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
	at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:833)
```

### Version

21.1.1

### Expected behavior

I can login to the admin console.

### Actual behavior

I get error 500.

### How to Reproduce?

Go to my.server.com/admin.
Enter admin credentials.
Try to log in.

**Comment by LZirus** [comment]
I keep getting this exact error while using chrome. When I switch to firefox, everything works as expected.

**Comment by janinko** [comment]
For me, it's the same backend error, for chrom and firefox, I get 500 on:
https://my.server.com/admin/master/console/whoami

On firefox I get this in browser:
```
NetworkError@https://my.server.com/resources/i3n4n/admin/keycloak.v2/assets/index-d2c499e7.js:67:43535
fetchWithError@https://my.server.com/resources/i3n4n/admin/keycloak.v2/assets/index-d2c499e7.js:67:43710
async*requestWithParams@https://my.server.com/resources/i3n4n/admin/keycloak.v2/assets/index-d2c499e7.js:67:46307
async*request/<@https://my.server.com/resources/i3n4n/admin/keycloak.v2/assets/index-d2c499e7.js:67:44801
WhoAmIContextProvider/<@https://my.server.com/resources/i3n4n/admin/keycloak.v2/assets/index-d2c499e7.js:81:100512
useFetch/<@https://my.server.com/resources/i3n4n/admin/keycloak.v2/assets/index-d2c499e7.js:80:216488
Rj@https://my.server.com/resources/i3n4n/admin/keycloak.v2/assets/index-d2c499e7.js:40:24279
Ik@https://my.server.com/resources/i3n4n/admin/keycloak.v2/assets/index-d2c499e7.js:40:42289
Fk@https://my.server.com/resources/i3n4n/admin/keycloak.v2/assets/index-d2c499e7.js:40:36532
jg@https://my.server.com/resources/i3n4n/admin/keycloak.v2/assets/index-d2c499e7.js:38:3280
Sk@https://my.server.com/resources/i3n4n/admin/keycloak.v2/assets/index-d2c499e7.js:40:37082
rl@https://my.server.com/resources/i3n4n/admin/keycloak.v2/assets/index-d2c499e7.js:40:53149
sl@https://my.server.com/resources/i3n4n/admin/keycloak.v2/assets/index-d2c499e7.js:40:53332
reactDom_production_min.render@https://my.server.com/resources/i3n4n/admin/keycloak.v2/assets/index-d2c499e7.js:40:57034
@https://my.server.com/resources/i3n4n/admin/keycloak.v2/assets/index-d2c499e7.js:81:227687
```

**Comment by ssilvert** [comment]
I don't think we can really call this a UI issue since it's getting NPE on the server side.   Switching to core team.

**Comment by mposolda** [comment]
This might be a valid issue, but Keycloak team probably doesn't have time to look at this in the near future. Anyone is welcome to send the PR with fixing it. See the "Contributors" section here https://www.keycloak.org/community for the details on how to send the PR for this.

**Comment by hossain-sazzad** [comment]
Reproduced the error. I am working on this issue. I will send PR asap.

**Comment by hossain-sazzad** [comment]
> The NPE is coming from `AdminConsole.java:230` which is `createRealm = user.hasRole(masterRealm.getRole(AdminRoles.CREATE_REALM));` I was able to get the error when I removed `create-realm` role from master realm. So maybe this is a misconfiguration issue?
> 
> Maybe we could improve the UI and add a `null` guard and return 403. But I let the core/UI team to decide.

Do you mean to restrict user from removing CREATE_REALM role from master realm?

**Comment by Tomisin-Rova** [comment]
> Reproduced the error. I am working on this issue. I will send PR asap.

hi, i'm still getting this error in v21.1.1

**Comment by saguntumkar** [comment]
Hi There...any updates on this issue? I am still getting this error in 21.1.1

**Comment by hossain-sazzad** [comment]
@mposolda could you please review the pr? Or please let me know whom to tag?

**Comment by NagenderPulluri** [comment]
Any resolution for this ? it's a bad experience for the end user to see that Unexpected Application Error.

**Comment by Blunderchips** [comment]
Any plans to solve this? We are unable to login into our keycloak environment.

**Comment by Blunderchips** [comment]
> 

Also returning `500` on firefox.

![image](https://github.com/keycloak/keycloak/assets/7081037/350f416a-da37-49c6-a334-cadc44ebd7b4)

**Comment by Blunderchips** [comment]
> Do we know roughly how long it is going to take to get a new docker image pushed up to dockerhub?

I fail to see how the releasing of a new version containing the fix is regarded as off-topic to the fix. 
Are there any plans to publish this fix to dockerhub?

**Comment by jonkoops** [comment]
@Blunderchips this will released with 22.0.0, as the milestone indicated already. Bumping a thread with these comments creates unnecessary notifications for maintainers, and since they do not pertain to the issue itself they will be marked as off-topic.
